src/libraries/System.Text.RegularExpressions/tests/System.Text.RegularExpressions.Generators.Tests/RegexGeneratorParserTests.cs

```diff
@@ -0,0 +1,432 @@
+                                    {
+                                        [RegexGenerator(""ab"")]
+                                        private static partial Regex Valid();
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            ", compile: true));
+        }
+
+        [Fact]
+        public async Task Valid_NullableRegex()
+        {
+            Assert.Empty(await RunGenerator(@"
+                #nullable enable
+                using System.Text.RegularExpressions;
+                partial class C
+                {
+                    [RegexGenerator(""ab"")]
+                    private static partial Regex? Valid();
+                }
+            ", compile: true));
+        }
+
+        [Fact]
+        public async Task Valid_InternalRegex()
+        {
+            Assert.Empty(await RunGenerator(@"
+                using System.Text.RegularExpressions;
+                partial class C
+                {
+                    [RegexGenerator(""ab"")]
+                    internal static partial Regex Valid();
+                }
+            ", compile: true));
+        }
+
+        [Fact]
+        public async Task Valid_PublicRegex()
+        {
+            Assert.Empty(await RunGenerator(@"
+                using System.Text.RegularExpressions;
+                partial class C
+                {
+                    [RegexGenerator(""ab"")]
+                    public static partial Regex Valid();
+                }
+            ", compile: true));
+        }
+
+        [Fact]
+        public async Task Valid_PrivateProtectedRegex()
+        {
+            Assert.Empty(await RunGenerator(@"
+                using System.Text.RegularExpressions;
+                partial class C
+                {
+                    [RegexGenerator(""ab"")]
+                    private protected static partial Regex Valid();
+                }
+            ", compile: true));
+        }
+
+        [Fact]
+        public async Task Valid_PublicSealedClass()
+        {
+            Assert.Empty(await RunGenerator(@"
+                using System.Text.RegularExpressions;
+                public sealed partial class C
+                {
+                    [RegexGenerator(""ab"")]
+                    private static partial Regex Valid();
+                }
+            ", compile: true));
+        }
+
+        [Fact]
+        public async Task Valid_InternalAbstractClass()
+        {
+            Assert.Empty(await RunGenerator(@"
+                using System.Text.RegularExpressions;
+                internal abstract partial class C
+                {
+                    [RegexGenerator(""ab"")]
+                    private static partial Regex Valid();
+                }
+            ", compile: true));
+        }
+
+        [Fact]
+        public async Task Valid_MultiplRegexMethodsPerClass()
```

Comment:
```suggestion
        public async Task Valid_MultipleRegexMethodsPerClass()
```
As a temporary optimization, would it make sense to include all the "positive case" regex generator tests above into a single file/single test? (this one basically).. would that speed up the tests noticeably?

Reply:
and maybe combine some/all of the failing ones, assuming the compiler will return all the diagnostics and we can check the line numbers match expectations.

Reply:
Never mind, I see we're running the generator on far more patterns. Still I wonder whether combining might be the path to more acceptable perf.

Reply:
> would that speed up the tests noticeably?

No.  These tests are very fast.  What's adding all the time is we're now running literally thousands of regexes as part of the main test suite, each of which is running the generator, compiling an assembly, etc.